GPU/Math3D.h

```diff
@@ -1363,6 +1377,59 @@ inline Vec3<float> Vec3<float>::operator * (const float &other) const {
 	return Vec3<float>(_mm_mul_ps(SAFE_M128(vec), _mm_set_ps1(other)));
 }
 
+// Vec4<int> operation
+template<>
+inline Vec4<int> Vec4<int>::operator + (const Vec4 &other) const {
+	return Vec4<int>(_mm_add_epi32(SAFE_M128I(ivec), SAFE_M128I(other.ivec)));
+}
+
+template<>
+inline Vec4<int> Vec4<int>::operator * (const Vec4 &other) const {
+	__m128i a = SAFE_M128I(ivec);
+	__m128i b = SAFE_M128I(other.ivec);
+#if _M_SSE >= 0x401
+	return Vec4<int>(_mm_mullo_epi32(a, b));
```

Comment:
Typically, people are using builds that weren't self-compiled for their target, so our current SSE4 usage is all runtime detected.  I think most Linux builds are actually not with SSE4, so this wouldn't be used unless you go the `[[gnu::target("sse4.1")]]` path.

Windows does define `_M_SSE` to 0x402 (statically iirc, even when SSE4 is not present.)  So this would probably cause an illegal instruction if anyone ran it on an older Windows device... though maybe we don't even officially support a Windows version that can run on a processor with SSE 4 anymore.

-[Unknown]

Reply:
I definitely don't want to introduce unguarded SSE4 into win builds. I suppose I'll stick with SSE2-only path here, and the code which wants to use `_mm_mullo_epi32` will have to do so explicitly.

On Linux even though `__SSE2__` etc. are "magic" for `[[gnu::target(...)]]`, I don't think that translates to `_M_SSE`.

Reply:
Right, we normally now just use the presence of _M_SSE to indicate we can runtime check newer.  We could in theory use cpu_info.bSSE4_1, but not sure how worth it it'd be here.  Also, I'd wish that any compiler told to compile for SSE4+ might recognize this sequence and ftfy to an sse4 multiply, it's pretty common.

-[Unknown]